global-api/routes/city_locode_endpoint.py

```diff
@@ -0,0 +1,135 @@
+from fastapi import FastAPI
+from sqlalchemy import create_engine, text 
+from sqlalchemy.orm import sessionmaker
+
+app = FastAPI()
+
+# Read the DATABASE_URL from the .env file
+DATABASE_URL = config("DATABASE_URL")
+
+# Create a SQLAlchemy engine for database connection
+engine = create_engine(DATABASE_URL)
+
+# Create a session maker for SQLAlchemy
+SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+# Dictionary to map gpc_sector to reference_number
+gpc_sector_to_reference_number = {
+    "stationary_energy":["I.4.1"],
+    "transportation":["II.1.1","II.4.1", "II.4.3"],
+    "waste":["III.1.2"],
+    "IPPU": ["IV.1.1"],
+    "AFOLU": ["V.3.1", "V.1.1","V.2.1"],
+}
+
+def sector_to_filename(inventoryPart):
+    gpc_reference_numbers = gpc_sector_to_reference_number.get(inventoryPart, [])
+    filenames = []
+
+    with SessionLocal() as session:
+        for reference_number in gpc_reference_numbers:
+            query_filenames = text("SELECT DISTINCT(filename) FROM ? WHERE reference_number = :reference_number")
+            filenames_result = session.execute(query_filenames, {"reference_number": reference_number}).fetchall()
+            filenames.extend(row[0] for row in filenames_result)
+
+    return filenames
+
+def db_query(filenames, locode, year):
+    results = []
+
+    with SessionLocal() as session:
+        for filename in filenames:
+            query = text(f"SELECT * FROM {filename} "
+                        f"WHERE locode = :locode "
+                        f"AND year = :year")
```

Comment:
This query expects that we have the locode for all entries in the DB. If this ends up being problematic in the import step, I would suggest storing lat, lng coordinates as a PostGIS point so we could query it by the GeoJSON polygon of the city.

If this works out it could be much more efficient to query though 👍

Reply:
@lemilonkh we will have the locode in the `asset` table. I'll work with @mfonsecaOEF and @lgloege to make that happen.